app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -147,11 +151,20 @@ class UserSession @Inject constructor(
   }
 
   fun loggedInUser(): Observable<Optional<LoggedInUser>> {
-    return loggedInUserPreference.asObservable()
+    return appDatabase.userDao().user()
+        .toObservable()
+        .map {
+          when {
```

Comment:
Can we use an `if-else` expression here since there are only two cases.

Reply:
Sure done